src/components/store/Navbar/index.tsx

```diff
@@ -0,0 +1,39 @@
+import Diamonds from '@/components/store/Diamonds';
+import { config } from '@/lib';
+import BackChevron from '@/public/assets/icons/back-chevron.svg';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import styles from './style.module.scss';
+
+interface NavbarProps {
+  balance: number;
+  showBack?: boolean;
+}
+
+const Navbar = ({ balance, showBack }: NavbarProps) => {
+  const router = useRouter();
+
+  return (
+    <nav className={styles.navbar}>
+      {showBack && (
+        <button type="button" className={styles.back} onClick={() => router.back()}>
```

Comment:
My original approach was that each page would provide the URL to return to as a prop, so a collection page would pass the storefront page, an item page could pass a collection URL/the storefront page, etc.

Raymond suggested using `history.back()`. Ideally, I'd hide the button if the previous page isn't inside `/store`, e.g. the user came from another part of the ACM website or outside.

A partial solution is using `document.referrer`, but NextJS's router to some degree acts like a SPA, so the page never reloads, and the value of `document.referrer` stays the same while the user navigates between pages on the website.

NextJS router [doesn't provide components direct access to the history](https://github.com/vercel/next.js/discussions/36723) of routes within the app, and tracking paths seems involved and potentially buggy since you'd also have to track when the user uses the browser's forward and back buttons.

My personal opinion is that the back button doesn't add any functionality the user doesn't already have (the browser already has a back button), and it adds confusion because the mental model for a back button inside an app vs. a website is very different (I'd expect clicking on a store icon then going back goes to the previous page on a website, and the home page in an app). I think that we should remove the back button altogether, or replace it with breadcrumbs `Store > Collection name > Item name`.